ddapm_test_agent/agent.py

```diff
@@ -308,6 +309,23 @@ async def handle_session_traces(self, request: Request) -> web.Response:
         traces = await self._traces_by_session(token)
         return web.json_response(traces)
 
+    async def handle_session_requests(self, request: Request) -> web.Response:
+        token = request["session_token"]
+        resp = []
+        for req in self._requests_by_session(token):
+            if req.match_info.handler not in (
+                self.handle_v04_traces,
+                self.handle_v05_traces,
+            ):
+                continue
+            resp.append(
+                {
+                    "headers": dict(req.headers),
```

Comment:
include url + method or any other info here?

Reply:
we could, not sure if it's relevant for trace client testing scenarios. This could easily be added in a follow-up 🤷‍♂️